### DISCUSSION TITLE
How to change a data type of a column with existing data

### DISCUSSION THREAD

**Discussion by ziolkowskiux** [supabase/supabase#19064] [Questions] [answered]
Hi,
I am trying to change a column that has a data type `int8` to `uuid` but Supabase throws a following error:
`failed to update pg.columns with the given ID: cannot cast type bigint to uuid`
Is there any way to change data type? I don't mind if the data in the column will be overwritten. I think it is also worth to mention that I am a complete beginner when it comes to databases 🙂.

Thank you in advance,
Maciej

**Comment by ssudaraka** [comment, marked as answer]
You can't directly convert an int column to uuid because the default value will be not valid with the uuid column. You can try something suggested in this Stack Overflow thread - https://stackoverflow.com/questions/20342717/postgresql-change-column-type-from-int-to-uuid.
